app/controllers/api/v1/Web.java

```diff
@@ -1061,6 +1072,276 @@ public static Result restApplicationFromApplicationId(String applicationid) {
+      JsonArray heuristics = new JsonArray();
+
+      for (JobType jobtype : applicationTypeListMap.get(key)) {
+        JsonObject jobTypeNode = new JsonObject();
+        jobTypeNode.addProperty(JsonKeys.NAME, jobtype.getName());
+        jobTypes.add(jobTypeNode);
+      }
+
+      for (Heuristic heuristic : ElephantContext.instance().getHeuristicsForApplicationType(key)) {
+        JsonObject heuristicNode = new JsonObject();
+        heuristicNode.addProperty(JsonKeys.NAME, heuristic.getHeuristicConfData().getHeuristicName());
+        heuristics.add(heuristicNode);
+      }
+
+      applicationType.addProperty(JsonKeys.NAME, key.getName());
+      applicationType.add(JsonKeys.JOB_TYPES, jobTypes);
+      applicationType.add(JsonKeys.HEURISTICS, heuristics);
+      jobCategory.add(applicationType);
+    }
+
+    for (Severity severity : Severity.values()) {
+      JsonObject severityObject = new JsonObject();
+      severityObject.addProperty(JsonKeys.NAME, severity.getText());
+      severityObject.addProperty(JsonKeys.VALUE, severity.getValue());
+      severities.add(severityObject);
+    }
+
+    searchOptions.add(JsonKeys.JOB_CATEGORY, jobCategory);
+    searchOptions.add(JsonKeys.SEVERITIES, severities);
+    searchOptions.addProperty(JsonKeys.ID, "search");
+    JsonObject parent = new JsonObject();
+    parent.add(JsonKeys.SEARCH_OPTS, searchOptions);
+    return ok(new Gson().toJson(parent));
+  }
+
+  /**
+   * Returns the search results for the given query
+   * @return
+   * JsonObject:
+   *
+   * <pre>
+   *   {
+   *         search-results: {
+   *         id: "id"
+   *         start: 0,
+   *         end: 20,
+   *         total: 0,
+   *         summaries: [
+   *                  {
+   *                    application_summary_object
+   *                  }
+   *                ]
+   *          }
+   *  }
+   * </pre>
+   */
+  public static Result search() {
+    DynamicForm form = Form.form().bindFromRequest(request());
+    JsonObject parent = new JsonObject();
+
+    int startIndex = SEARCH_DEFAULT_PAGE_START_INDEX;
+    int offset = SEARCH_DEFAULT_PAGE_OFFSET;
+    int end = 0;
+    int total = 0;
+
+    if (form.get("start") != null && form.get("start") != "") {
+      startIndex = Integer.valueOf(form.get("start"));
+    }
+    if (form.get("offset") != null && form.get("offset") != "") {
+      offset = Integer.valueOf(form.get("offset"));
+    }
+
+    if (startIndex < 0) {
+      startIndex = 0;
+    }
+
+    if (offset > SEARCH_APPLICATION_MAX_OFFSET) {
+      offset = SEARCH_APPLICATION_MAX_OFFSET;
+    } else if (offset <= 0) {
+      return ok(new Gson().toJson(parent));
+    }
+
+    Query<AppResult> query =
+        Application.generateSearchQuery(AppResult.getSearchFields(), Application.getSearchParams());
+
+    total = query.findRowCount();
+
+    if (startIndex > total) {
+      startIndex = total;
+    }
+
+    List<AppResult> results = query.setFirstRow(startIndex).setMaxRows(offset)
```

Comment:
According to mysql convention, we use OFFSET and LIMIT, where OFFSET is the initial start point and LIMIT is the number of records to fetch.

Here you seem to be using ```offset``` as limit which can be confusing. Can you use OFFSET and LIMIT in place of startIndex and offset.